File: packages/format/src/h264/failure.ts
import { FormatError } from "../errors.js";

export function h264Invalid(
  path: string,
  message: string,
  offset?: number
): never {
  throw new FormatError("PROFILE_INVALID", message, {
    path,
    ...(offset === undefined ? {} : { offset })
  });
}
export function requireH264(
  condition: boolean,
  path: string,
  message: string,
  offset?: number
): asserts condition {
  if (!condition) {
    h264Invalid(path, message, offset);
  }
}

File: flutter/packages/aval_format/lib/src/h264/failure.dart
/// Shared H264 subsystem failure helper.
///
/// Dart port of `packages/format/src/h264/failure.ts`.
library;

import '../errors.dart';

/// Port of `h264Invalid` (`failure.ts:3`).
Never h264Invalid(String path, String message, [int? offset]) {
  throw FormatError(
    FormatErrorCode.profileInvalid,
    message,
    FormatErrorDetails(path: path, offset: offset),
  );
}

/// Port of `requireH264` (`failure.ts:13`).
void requireH264(
  bool condition,
  String path,
  String message, [
  int? offset,
]) {
  if (!condition) {
    h264Invalid(path, message, offset);
  }
}
